### PR TITLE
Fix #12511: Datatable/TreeTable filterNormalize

### LINKED ISSUE #12511: DataTable/TreeTable: New `filterNormalize` attribute
### Description

Firstly I really appreciate the new `filterNormalize="true"` attribute, available in components like `<p:selectCheckboxMenu/>`
 
 **I was wondering if it is possible to have the same behaviour in _<p:datatable columns_ /> as well.** E.g.:
`<p:column   filterBy="#{op.name}"   filterMatchMode="contains"  **filterNormalize="true"**   .../> `

BTW, I'm actually using a custom filter method in Bean and using it like that: `filterFunction="#{utilBean.filterContainsWithoutAccents}"` and it works fine when user filter using the Column Header Filter Box, but fails when user use the main global filter box like:
` <h:inputText id="globalFilter"  onkeydown="if (event.keyCode == 13) { return  PF('tableWdg').filter(); }"        />` which not calls ServerSide Bean method` public boolean filterContainsWithoutAccents(Object value, Object filter, Locale locale) {.`.. anymore.

Regards 

### Describe the solution you would like

For me, the cleanest solution would be  to have  `filterNormalize=""` attribute like we have now in components like `<p:selectCheckboxMenu/>` to filtering for column data, escaping accented characters.

As a workaround I'm using a server side custom filter method called in: `filterFunction="#{utilBean.filterContainsWithoutAccents}"`  but apparently it just work   when called by the 'Column Header Filter Box', and it's not   called   by global filter `PF('tableWdg').filter();`,  there is a way to call all declared columns `filterFunction=""` in that context as well?

Thanks in advance.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by melloware**
Fix #12511: Datatable/TreeTable filterNormalize

@jepsar this is what I was thinking because its easier to capture here on the server side no matter what filters or global filters in play.

### CODE REVIEW

File: primefaces/src/main/java/org/primefaces/component/datatable/feature/FilterFeature.java [hunk 1]
```diff
@@ -138,7 +141,7 @@ public void filter(FacesContext context, DataTable table) {
 
                 if (globalFilter != null && globalFilter.isActive() && !globalMatch.get() && !hasGlobalFilterFunction) {
                     FilterConstraint constraint = globalFilter.getConstraint();
-                    Object filterValue = globalFilter.getFilterValue();
+                    Object filterValue = LangUtils.normalize(globalFilter.getFilterValue(), table.isFilterNormalize());
                     globalMatch.set(constraint.isMatching(context, columnValue, filterValue, filterLocale));
```
Comment (jepsar) [inline primefaces/src/main/java/org/primefaces/component/datatable/feature/FilterFeature.java]: We should normalize the columnValue as well, right?

Comment (melloware) [reply primefaces/src/main/java/org/primefaces/component/datatable/feature/FilterFeature.java]: Good point. We probably should for comparison.

**Comment by Rapster** [comment]
Shouldn't we handle this in FilterMeta instead?

**Comment by melloware** [comment]
yes in theory we could i have to see where those FilterMeta are constructed and would they have access to the `table.isFilterNormalize()` new setting for both DT and TT?

**Comment by Rapster** [comment]
You can either pass UITable or add a boolean to `of` method

**Comment by Rapster** [comment]
Missing code in setFilterValue and getLocalValue ?

**Comment by melloware** [comment]
All i did was when `FilterMeta.of` is called to create filter meta is passes the DT or TT value for `isFilterNormalize()`.  And then we apply the normalize on the value.  Is this wrong?

**Comment by jepsar** [comment]
If the dataTable contain "café" but you search on "cafe", I believe you now only normalize "cafe".

**Comment by melloware** [comment]
so was my previous PR correct?  where i did everything last minute in FilterFeature?

File: primefaces/src/main/java/org/primefaces/component/datatable/feature/FilterFeature.java (hunk 1 above)
Comment (jepsar) [reply primefaces/src/main/java/org/primefaces/component/datatable/feature/FilterFeature.java]: @melloware basically this was the same issue I believe.

**Comment by melloware** [comment]
So we need to also Normalize the other side of the filtering where its checking the value.  I have to figure out where that happens 😄

**Comment by melloware** [comment]
OK i see what @Rapster meant needed to add it to `SetFilterValue` and `getLocalValue` and now its comparing the filter to the table value properly.

**Comment by tandraschko** [comment]
i would personally prefer to not put logic into setters or getters as we loose the actual filterValue (also if not needed currently) and create a method like: 'resolveFilterValue', where we can put logic into 

it would be just a bit nicer IMO but the PR is also ok for me.

**Comment by melloware** [comment]
@Rapster any thoughts on the resolveFilterValue?

**Comment by Rapster** [comment]
Code is fine IMO

**Pull request merged**